crates/walrus-simtest/tests/simtest_event_blob.rs

```diff
@@ -168,6 +285,49 @@ mod tests {
         blob_info_consistency_check.check_storage_node_consistency();
     }
 
+    /// This test verifies that the node can correctly recover from a forked event blob.
+    #[ignore = "ignore integration simtests by default"]
+    #[walrus_simtest]
+    async fn test_event_blob_local_fork_recovery() {
+        let (_sui_cluster, mut walrus_cluster, client, _) =
+            test_cluster::E2eTestSetupBuilder::new()
+                .with_epoch_duration(Duration::from_secs(15))
+                .with_test_nodes_config(TestNodesConfig {
+                    node_weights: vec![2, 2, 3, 3, 3],
+                    ..Default::default()
+                })
+                .with_num_checkpoints_per_blob(20)
+                .with_communication_config(
+                    ClientCommunicationConfig::default_for_test_with_reqwest_timeout(
+                        Duration::from_secs(2),
+                    ),
+                )
+                .build_generic::<SimStorageNodeHandle>()
+                .await
+                .unwrap();
+
+        let client = Arc::new(client);
+
+        // Run workload to get some event blobs certified
+        tokio::time::sleep(Duration::from_secs(30)).await;
+
+        // Restart nodes with different checkpoint numbers to create fork
+        restart_node_with_checkpoints(&mut walrus_cluster, 0, |i| 30 + i as u32).await;
+
+        // Wait for event blob certification to get stuck
+        wait_for_event_blob_writer_to_fork(&mut walrus_cluster, &client, 0)
+            .await
+            .unwrap();
+
+        // Restart nodes with same checkpoint number to recover
+        restart_node_with_checkpoints(&mut walrus_cluster, 0, |_| 20).await;
+
+        // Verify recovery
+        wait_for_node_to_recover(&walrus_cluster.nodes[0])
```

Comment:
Nit: s/wait_for_node_to_recover/wait_for_event_blob_writer_to_recover

node to recover may be misleading since nodes also have a recovery mode to recover processing events.